packages/@aws-cdk/aws-eks/test/test.cluster.ts

```diff
@@ -188,6 +188,30 @@ export = {
 
   },
 
+  'spot interrupt handler is not added if addSpotInterruptHandler is false when connecting self-managed nodes'(test: Test) {
+
+    // GIVEN
+    const { stack, vpc } = testFixture();
+    const cluster = new eks.Cluster(stack, 'Cluster', {
+      vpc,
+      defaultCapacity: 0,
+      version: CLUSTER_VERSION,
+      prune: false,
+    });
+
+    const selfManaged = new asg.AutoScalingGroup(stack, 'self-managed', {
+      instanceType: new ec2.InstanceType('t2.medium'),
+      vpc: vpc,
+      machineImage: new ec2.AmazonLinuxImage(),
+    });
+
+    // WHEN
+    cluster.connectAutoScalingGroupCapacity(selfManaged, { addSpotInterruptHandler: false });
+
+    expect(stack).notTo(haveResource(eks.HelmChart.RESOURCE_TYPE));
```

Comment:
This will start failing if we add an unrelated default helm chart to our cluster. Make it a little more precise by asserting on the specific chart id:

```ts
test.equal(cluster.node.findAll().filter(c => c.node.id === 'chart-spot-interrupt-handler').length, 0);
```